### PR TITLE
feat(workspace): the workspace support power action and add status attribute

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

API and console are support power action.
Add power action function for the Workspace desktop.
+ add **power_action** and **power_action_type** parameters.
+ add **status** attribute.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Add power action function for the Workspace desktop.
2. Add a status attribute.

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1990.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1990.315s

make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_powerAction'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_powerAction -timeout 360m -parallel 4
=== RUN   TestAccDesktop_powerAction
=== PAUSE TestAccDesktop_powerAction
=== CONT  TestAccDesktop_powerAction
--- PASS: TestAccDesktop_powerAction (1496.10s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1496.135s
```

### CODE REVIEW

File: docs/resources/workspace_desktop.md [hunk 1]
```diff
@@ -122,6 +124,17 @@ The following arguments are supported:
...
+* `power_action` - (Optional, String) Specifies the power action to be done for the desktop.
+  The valid values are **os-start**, **os-stop**, **reboot**, **os-hibernate**.
+
+  -> The `power_action` is a one-time action.
+     Currently , only Windows is supported for `os-hibernate`.
+  
+* `power_action_type` - (Optional, String) Specifies the power action mechanisms for the desktop.
+  The valid values are as follows:
+  + **SOFT**: Normal operation.
+  + **HARD**: Forced operation.
```
Comment (Lance52259) [inline docs/resources/workspace_desktop.md:134]: When `power_action_type` is updated, the corresponding action is also executed. This behavior is not described in the document.

File: docs/resources/workspace_desktop.md [hunk 2]
```diff
@@ -200,7 +215,7 @@ resource "huaweicloud_workspace_desktop" "test" {
 
   lifecycle {
     ignore_changes = [
-      user_email, nic,
+      user_email, "power_action","power_action_type",
```
Comment (Lance52259) [inline docs/resources/workspace_desktop.md]:
```suggestion
      user_email, power_action, power_action_type,
```
The quotation marks here are superfluous

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_powerAction'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_powerAction -timeout 360m -parallel 4
=== RUN   TestAccDesktop_powerAction
=== PAUSE TestAccDesktop_powerAction
=== CONT  TestAccDesktop_powerAction
--- PASS: TestAccDesktop_powerAction (1578.35s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1578.476s

File: docs/resources/workspace_desktop.md (hunk 2 above)
Comment (wuzhuanhong) [reply docs/resources/workspace_desktop.md]: fixed

File: docs/resources/workspace_desktop.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/resources/workspace_desktop.md:134]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_powerAction'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_powerAction -timeout 360m -parallel 4
=== RUN   TestAccDesktop_powerAction
=== PAUSE TestAccDesktop_powerAction
=== CONT  TestAccDesktop_powerAction
--- PASS: TestAccDesktop_powerAction (1338.82s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1338.865s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_powerAction'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_powerAction -timeout 360m -parallel 4
=== RUN   TestAccDesktop_powerAction
=== PAUSE TestAccDesktop_powerAction
=== CONT  TestAccDesktop_powerAction
--- PASS: TestAccDesktop_powerAction (1390.16s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1390.205s

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go [hunk 3]
```diff
@@ -607,6 +629,81 @@ func updateDesktopNetwork(ctx context.Context, client *golangsdk.ServiceClient,
...
+			"os-stop":      "SHUTOFF",
+			"reboot":       "ACTIVE",
+			"os-hibernate": "SHUTOFF",
+		}
+
+		powerAction := d.Get("power_action").(string)
+		// TaskStatus variable is always an empty string when the desktop power action is completed.
+		// If the desktop power action changes from one state to another, taskStatus is an empty string for a long time,
+		// whether a desktop action is completed cannot be determined only by taskStatus.
+		if taskStatus == "" && resp.Status == statusMap[powerAction] {
```
Comment (ShiChangkuo) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]: please define taskStatus as near as it was used, or using `resp.TaskStatus` directly
```suggestion
		if resp.TaskStatus == "" && resp.Status == statusMap[powerAction] {
```

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go [hunk 4]
```diff
@@ -154,6 +155,91 @@ func TestAccDesktop_UpdateWithEpsId(t *testing.T) {
...
+					resource.TestCheckResourceAttr(resourceName, "status", "SHUTOFF"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"delete_user",
+					"image_type",
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go:232]: why are `delete_user`, `image_type` and `vpc_id` missed in Import section of docs?

Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go:232]: fixed

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go (hunk 3 above)
Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]: fixed

File: docs/resources/workspace_desktop.md [hunk 5]
```diff
@@ -122,6 +122,17 @@ The following arguments are supported:
   operations.  
   Defaults to **false**. Changing this will create a new resource.
 
+* `power_action` - (Optional, String) Specifies the power action to be done for the desktop.
+  The valid values are **os-start**, **os-stop**, **reboot**, **os-hibernate**.
+
+  -> The `power_action` is a one-time action.
+     Currently , only Windows is supported for `os-hibernate`.
```
Comment (ShiChangkuo) [inline docs/resources/workspace_desktop.md]:
```suggestion
     Currently, only Windows supports **os-hibernate** action.
```

Comment (wuzhuanhong) [reply docs/resources/workspace_desktop.md]: fixed

**Pull request merged**